service/history/shard/context_impl.go

```diff
@@ -206,11 +206,11 @@ func (s *ContextImpl) AssertOwnership(
 	s.wLock()
 	defer s.wUnlock()
 
-	if err := s.errorByState(); err != nil {
-		return err
+	if s.shardInfo == nil {
```

Comment:
could shardInfo be null in AssertOwnership?

Reply:
yes, especially during shard initial startup

Reply:
the idea is to make AssertOwnership not blocked by "shard status unknown"